src/v/cloud_storage/segment_chunk_api.cc

```diff
@@ -226,6 +179,67 @@ ss::future<> segment_chunks::trim_chunk_files() {
     _eviction_timer.rearm(_eviction_jitter());
 }
 
+ss::future<> segment_chunks::run_hydrate_bg() {
+    auto g = _gate.hold();
+    vassert(_started, "chunk API is not started");
+
+    while (!_gate.is_closed()) {
+        try {
+            co_await _bg_cvar.wait(
+              [this] { return downloads_in_progress() || _gate.is_closed(); });
+            // TODO prioritize "real" requests over prefetches
+            co_await ss::max_concurrent_for_each(
```

Comment:
Doesn't this cause a lot (too much) throw away work?

Reply:
Do you mean because of the `max_concurrent_for_each`? Or because we do not filter for chunks with empty waiters before entering the method to hydrate?

Reply:
Yea i guess it wouldn't be tough to get only the chunks that are `download_in_progress` and then do this max_concurrent call? I suppose the condition variable condition means there definitely will be at least one waiter, but it seems like an easy win to filter up front

Reply:
Added a filter